### ISSUE TITLE
Split repo

### ISSUE THREAD

**Issue by MGatner** [heroespatchnotes/heroes-talents#79]
@stuaroo I've been thinking about repo maintenance. We're getting big, almost 500mb. I've been thinking it might be good to split the **raw** folder and contents out into its own repo - still maintained the same way but we could retroactively trim it out of this repo to cut down on the size, and it would make diffs between patches much easier to read because it would be just the JSON files and any update icons. I'll check with @koliva8245 if he would like to host this, or have it hosted by heroespatchnotes, but let me know what you think.

Everyone else: do you use the **raw** folder currently? If so, would you use it from another source? Would you like the format changed at all?

**Comment by MGatner** [comment]
FYI it sounds like @koliva8245 has actually been having similar thoughts, and likely we could collaborate on a "raw" repo instead of trying to host it merged here: https://github.com/koliva8245/HeroesDataParser/issues/57

**Comment by stuaroo** [comment]
That sounds like a fine idea to me.  I sometimes use the raw output, but it will be just as easy to navigate to a different repo to grab it (and I can always run HeroesDataParser if I need it in a different format).

It will be nice to have easier/smaller diffs too; lately I've ended up cloning PRs to local to review them as my browser has had issues with our large diffs!

**Comment by MGatner** [comment]
Koliva has the new repos up:
* https://github.com/HeroesToolChest/heroes-data
* https://github.com/HeroesToolChest/heroes-images

If you're alright with it I will do a destructive repo prune of all the raw data history on my branch and then invite you to review it. If everything looks good we can force it on this repo - it will probably cause some havoc on any forks but I think the end result will be a lot cleaner and smaller, and we can focus on the formatted JSON files that makes this repo distinct.

**Comment by stuaroo** [comment]
Sounds good.  I've been browsing through those repos you linked and I guess I didn't notice this but it looks like the game strings are all using the default/description 0.  Are you thinking this repo's workflow is going to stay about the same, where you'll need to run HDP manually for every patch?  I thought the new repos might replace a step, but it seems like it'll still need to be run with --description 3 to get output with scaling.

**Comment by MGatner** [comment]
No, that will be part of the formatting, "nicifying" those descriptions. It will take a me a bit to update the in/out pipeline for the new repo and format so for now I will probably keep doing my own parsing, but this will keep me from having to sync that to this repo. And a eventually it will save a step.

**Comment by MGatner** [comment]
Exciting times! koliva has the data repos up (check out https://github.com/HeroesToolChest) which paves the way for us to split off the raw files and focus on the formatted versions.

To this end I rewrote my formatting scripts to use the **heroes-data** repo and posted them: https://github.com/tattersoftware/heroes-convert. There is still work to be done, mostly with the Filters and Interventions, but it's functional and I think a great step for this repo.

**Comment by MGatner** [comment]
@stuaroo We're ready to drop the raw content. I'm going to create a new issue with a warning since forks will all need to be force updated. This will includes links to my new fork with the split so you can review it. I'll keep my current fork around for historic purposes just in case.

**Comment by MGatner** [comment]
All set: https://github.com/heroespatchnotes/heroes-talents/issues/88. Stu please let me know when you'd like to pull the trigger on this.

**Comment by stuaroo** [comment]
Ok, feel free to go ahead at your convenience!

Just for my own git education, what commands/steps are you planning to run to prune the `raw` folder from the commit history?

**Comment by MGatner** [comment]
I used `filter-branch` and ran it against `master` and `ptr` separately:

`git filter-branch --force --index-filter "git rm --cached --ignore-unmatch -r raw/" --prune-empty --tag-name-filter cat -- --all`

After running the filters and compressing objects (`gc`) my local repo was 65mb (down from 450mb). The cloned GitHub version will probably more, at least until garbage collection kicks in, but it should still cut out a ton. The down side is that `filter-branch` basically “recommits” each commit so they have new IDs and aren’t compatible with the existing repo anymore. It does update tags though so we’ll keep all our versions.
